File: backend/schemas/cotizacion.py
from pydantic import BaseModel
from typing import Optional, List

class CotizacionBase(BaseModel):
    id: int
    fecha_hora: Optional[str]
    nombre: Optional[str]
    telefono: Optional[str]
    correo: Optional[str]
    proyecto: Optional[str]
    estilo: Optional[str]
    espacios: Optional[str]
    area_m2: Optional[float]
    habitaciones: Optional[int]
    tiempo: Optional[str]
    distrito: Optional[str]
    diseno: Optional[float]
    mobiliario: Optional[float]
    acabados: Optional[float]
    precio_final: Optional[float]
    created_at: Optional[str] = None

class ListResponse(BaseModel):
    total: int
    page: int
    page_size: int
    data: List[CotizacionBase]

class SummaryMetrics(BaseModel):
    total_cotizaciones: int
    suma_precio: float
    ticket_promedio: float
    m2_promedio: float

class SeriesPoint(BaseModel):
    x: str
    total: int
    suma_precio: float

class LabeledAgg(BaseModel):
    label: str
    total: int
    suma_precio: float
    promedio: float

File: frontend/src/types/Cotizacion.ts
export interface CotizacionRow {
id: number;
fecha_hora: string | null; // timestamptz
nombre: string | null;
telefono: string | null;
correo: string | null;
proyecto: string | null;
estilo: string | null;
espacios: string[] | null; // text[]
area_m2: number | null;
habitaciones: number | null;
tiempo: string | null;
distrito: string | null;
diseno: number | null;
mobiliario: number | null;
acabados: number | null;
precio_final: number | null;
created_at?: string | null;
};

// --- Tipos exportados para reutilizar en componentes ---
export type PaginatedResp<T> = {
  total: number;
  page: number;
  page_size: number;
  data: T[];
};

export type MonthlyPoint = {
  x: string;            // "YYYY-MM"
  total: number;        // conteo del mes
  suma_precio: number;  // suma de precio_final del mes
};

export type TopItem = {
  label: string;        // estilo o distrito
  total: number;
  suma_precio?: number;
  promedio?: number;
};

export type SummaryResp = {
  // Soporta ambos esquemas (dashboard/cotizaciones):
  total?: number;
  suma_total?: number;
  promedio?: number;

  total_cotizaciones?: number;
  suma_precio?: number;
  ticket_promedio?: number;
  m2_promedio?: number;
};
